### ISSUE TITLE
[bdev_verify_big_io] bdevperf: thread.c:2643: spdk_for_each_channel_continue: Assertion `rc == 0' failed.

### ISSUE THREAD

**Issue by mikeBashStuff** [spdk/spdk#3201]
# Sighting report

https://ci.spdk.io/results/autotest-per-patch/builds/112913/archive/rocky8-vg-autotest_45227/index.html

```
00:11:57.052  ************************************
00:11:57.052  START TEST bdev_verify_big_io
00:11:57.052  ************************************
00:11:57.053   08:52:52	-- common/autotest_common.sh@1084 -- # /home/vagrant/spdk_repo/spdk/build/examples/bdevperf --json /home/vagrant/spdk_repo/spdk/test/bdev/bdev.json -q 128 -o 65536 -w verify -t 5 -C -m 0x3 ''
00:11:57.310  [2023-11-28 08:52:52.898516] Starting SPDK v24.01-pre git sha1 055de83ac / DPDK 23.07.0 initialization...
00:11:57.310  [2023-11-28 08:52:52.898783] [ DPDK EAL parameters: bdevperf --no-shconf -c 0x3 --huge-unlink --no-telemetry --log-level=lib.eal:6 --log-level=lib.cryptodev:5 --log-level=user1:6 --iova-mode=pa --base-virtaddr=0x200000000000 --match-allocations --file-prefix=spdk_pid167724 ]
00:11:57.567  [2023-11-28 08:52:53.055436] app.c: 794:spdk_app_start: *NOTICE*: Total cores available: 2
00:11:57.824  [2023-11-28 08:52:53.323131] reactor.c: 937:reactor_run: *NOTICE*: Reactor started on core 0
00:11:57.824  [2023-11-28 08:52:53.323132] reactor.c: 937:reactor_run: *NOTICE*: Reactor started on core 1
00:11:58.390  [2023-11-28 08:52:53.839025] bdev.c:7953:bdev_open_ext: *NOTICE*: Currently unable to find bdev with name: Malloc1
00:11:58.390  [2023-11-28 08:52:53.839130] bdev.c:7953:bdev_open_ext: *NOTICE*: Currently unable to find bdev with name: Malloc1
00:11:58.390  [2023-11-28 08:52:53.846984] bdev.c:7953:bdev_open_ext: *NOTICE*: Currently unable to find bdev with name: Malloc2
00:11:58.390  [2023-11-28 08:52:53.847038] bdev.c:7953:bdev_open_ext: *NOTICE*: Currently unable to find bdev with name: Malloc2
00:11:58.390  [2023-11-28 08:52:53.855017] vbdev_passthru.c: 607:vbdev_passthru_register: *NOTICE*: Match on Malloc3
00:11:58.390  [2023-11-28 08:52:53.855096] bdev.c:7953:bdev_open_ext: *NOTICE*: Currently unable to find bdev with name: Malloc3
00:11:58.390  [2023-11-28 08:52:53.855133] vbdev_passthru.c: 731:bdev_passthru_create_disk: *NOTICE*: vbdev creation deferred pending base bdev arrival
00:11:58.390  [2023-11-28 08:52:54.070781] vbdev_passthru.c: 607:vbdev_passthru_register: *NOTICE*: Match on Malloc3
00:11:58.390  [2023-11-28 08:52:54.070896] vbdev_passthru.c: 636:vbdev_passthru_register: *NOTICE*: base bdev opened
00:11:58.390  [2023-11-28 08:52:54.070926] vbdev_passthru.c: 676:vbdev_passthru_register: *NOTICE*: io_device created at: 0x0x61600000ba80
00:11:58.390  [2023-11-28 08:52:54.070969] vbdev_passthru.c: 691:vbdev_passthru_register: *NOTICE*: bdev claimed
00:11:58.390  [2023-11-28 08:52:54.073100] vbdev_passthru.c: 704:vbdev_passthru_register: *NOTICE*: pt_bdev registered
00:11:58.390  [2023-11-28 08:52:54.073149] vbdev_passthru.c: 705:vbdev_passthru_register: *NOTICE*: created pt_bdev for: TestPT
00:11:58.958  [2023-11-28 08:52:54.489629] bdevperf.c:1744:bdevperf_construct_job: *WARNING*: Due to constraints of verify job, queue depth (-q, 128) can't exceed the number of IO requests which can be submitted to the bdev Malloc2p0 simultaneously (32). Queue depth is limited to 32
00:11:58.958  [2023-11-28 08:52:54.494005] bdevperf.c:1744:bdevperf_construct_job: *WARNING*: Due to constraints of verify job, queue depth (-q, 128) can't exceed the number of IO requests which can be submitted to the bdev Malloc2p0 simultaneously (32). Queue depth is limited to 32
00:11:58.958  [2023-11-28 08:52:54.499024] bdevperf.c:1744:bdevperf_construct_job: *WARNING*: Due to constraints of verify job, queue depth (-q, 128) can't exceed the number of IO requests which can be submitted to the bdev Malloc2p1 simultaneously (32). Queue depth is limited to 32
00:11:58.958  [2023-11-28 08:52:54.503929] bdevperf.c:1744:bdevperf_construct_job: *WARNING*: Due to constraints of verify job, queue depth (-q, 128) can't exceed the number of IO requests which can be submitted to the bdev Malloc2p1 simultaneously (32). Queue depth is limited to 32
00:11:58.958  [2023-11-28 08:52:54.508058] bdevperf.c:1744:bdevperf_construct_job: *WARNING*: Due to constraints of verify job, queue depth (-q, 128) can't exceed the number of IO requests which can be submitted to the bdev Malloc2p2 simultaneously (32). Queue depth is limited to 32
00:11:58.958  [2023-11-28 08:52:54.512990] bdevperf.c:1744:bdevperf_construct_job: *WARNING*: Due to constraints of verify job, queue depth (-q, 128) can't exceed the number of IO requests which can be submitted to the bdev Malloc2p2 simultaneously (32). Queue depth is limited to 32
00:11:58.958  [2023-11-28 08:52:54.517203] bdevperf.c:1744:bdevperf_construct_job: *WARNING*: Due to constraints of verify job, queue depth (-q, 128) can't exceed the number of IO requests which can be submitted to the bdev Malloc2p3 simultaneously (32). Queue depth is limited to 32
00:11:58.958  [2023-11-28 08:52:54.522084] bdevperf.c:1744:bdevperf_construct_job: *WARNING*: Due to constraints of verify job, queue depth (-q, 128) can't exceed the number of IO requests which can be submitted to the bdev Malloc2p3 simultaneously (32). Queue depth is limited to 32
00:11:58.958  [2023-11-28 08:52:54.526195] bdevperf.c:1744:bdevperf_construct_job: *WARNING*: Due to constraints of verify job, queue depth (-q, 128) can't exceed the number of IO requests which can be submitted to the bdev Malloc2p4 simultaneously (32). Queue depth is limited to 32
00:11:58.958  [2023-11-28 08:52:54.531162] bdevperf.c:1744:bdevperf_construct_job: *WARNING*: Due to constraints of verify job, queue depth (-q, 128) can't exceed the number of IO requests which can be submitted to the bdev Malloc2p4 simultaneously (32). Queue depth is limited to 32
00:11:58.958  [2023-11-28 08:52:54.535335] bdevperf.c:1744:bdevperf_construct_job: *WARNING*: Due to constraints of verify job, queue depth (-q, 128) can't exceed the number of IO requests which can be submitted to the bdev Malloc2p5 simultaneously (32). Queue depth is limited to 32
00:11:58.958  [2023-11-28 08:52:54.540294] bdevperf.c:1744:bdevperf_construct_job: *WARNING*: Due to constraints of verify job, queue depth (-q, 128) can't exceed the number of IO requests which can be submitted to the bdev Malloc2p5 simultaneously (32). Queue depth is limited to 32
00:11:58.958  [2023-11-28 08:52:54.545244] bdevperf.c:1744:bdevperf_construct_job: *WARNING*: Due to constraints of verify job, queue depth (-q, 128) can't exceed the number of IO requests which can be submitted to the bdev Malloc2p6 simultaneously (32). Queue depth is limited to 32
00:11:58.958  [2023-11-28 08:52:54.549306] bdevperf.c:1744:bdevperf_construct_job: *WARNING*: Due to constraints of verify job, queue depth (-q, 128) can't exceed the number of IO requests which can be submitted to the bdev Malloc2p6 simultaneously (32). Queue depth is limited to 32
00:11:58.958  [2023-11-28 08:52:54.554276] bdevperf.c:1744:bdevperf_construct_job: *WARNING*: Due to constraints of verify job, queue depth (-q, 128) can't exceed the number of IO requests which can be submitted to the bdev Malloc2p7 simultaneously (32). Queue depth is limited to 32
00:11:58.958  [2023-11-28 08:52:54.558261] bdevperf.c:1744:bdevperf_construct_job: *WARNING*: Due to constraints of verify job, queue depth (-q, 128) can't exceed the number of IO requests which can be submitted to the bdev Malloc2p7 simultaneously (32). Queue depth is limited to 32
00:11:58.958  [2023-11-28 08:52:54.664347] bdevperf.c:1744:bdevperf_construct_job: *WARNING*: Due to constraints of verify job, queue depth (-q, 128) can't exceed the number of IO requests which can be submitted to the bdev AIO0 simultaneously (78). Queue depth is limited to 78
00:11:58.958  [2023-11-28 08:52:54.672857] bdevperf.c:1744:bdevperf_construct_job: *WARNING*: Due to constraints of verify job, queue depth (-q, 128) can't exceed the number of IO requests which can be submitted to the bdev AIO0 simultaneously (78). Queue depth is limited to 78
00:11:59.217  Running I/O for 5 seconds...
00:12:05.840  
00:12:05.840                                                                                                  Latency(us)
00:12:05.840  
 Device Information          : runtime(s)       IOPS      MiB/s     Fail/s     TO/s    Average        min        max
00:12:05.840  
 Job: Malloc0 (Core Mask 0x1)
00:12:05.840  	 Verification LBA range: start 0x0 length 0x100
00:12:05.840  	 Malloc0             :       5.25     564.19      35.26       0.00     0.00  224642.83     310.13  818887.92
00:12:05.840  
 Job: Malloc0 (Core Mask 0x2)
00:12:05.840  	 Verification LBA range: start 0x100 length 0x100
00:12:05.840  	 Malloc0             :       5.30     511.00      31.94       0.00     0.00  248281.21     292.57  846849.95
00:12:05.840  
 Job: Malloc1p0 (Core Mask 0x1)
00:12:05.840  	 Verification LBA range: start 0x0 length 0x80
00:12:05.840  	 Malloc1p0           :       5.47     163.66      10.23       0.00     0.00  751037.45    1497.97 1086524.46
00:12:05.840  
 Job: Malloc1p0 (Core Mask 0x2)
00:12:05.840  	 Verification LBA range: start 0x80 length 0x80
00:12:05.840  	 Malloc1p0           :       5.40     284.30      17.77       0.00     0.00  435560.19    2496.61  731007.27
00:12:05.840  
 Job: Malloc1p1 (Core Mask 0x1)
00:12:05.840  	 Verification LBA range: start 0x0 length 0x80
00:12:05.840  	 Malloc1p1           :       5.69      92.72       5.80       0.00     0.00 1297931.15     791.89 1805548.01
00:12:05.840  
 Job: Malloc1p1 (Core Mask 0x2)
00:12:05.840  	 Verification LBA range: start 0x80 length 0x80
00:12:05.840  	 Malloc1p1           :       5.60      85.64       5.35       0.00     0.00 1411355.45     780.19 1973320.17
00:12:05.840  
 Job: Malloc2p0 (Core Mask 0x1)
00:12:05.840  	 Verification LBA range: start 0x0 length 0x20
00:12:05.840  	 Malloc2p0           :       5.42      79.64       4.98       0.00     0.00  381567.86     464.21  619159.16
00:12:05.840  
 Job: Malloc2p0 (Core Mask 0x2)
00:12:05.840  	 Verification LBA range: start 0x20 length 0x20
00:12:05.840  	 Malloc2p0           :       5.44      73.53       4.60       0.00     0.00  414487.14     464.21  659104.91
00:12:05.840  
 Job: Malloc2p1 (Core Mask 0x1)
00:12:05.840  	 Verification LBA range: start 0x0 length 0x20
00:12:05.840  	 Malloc2p1           :       5.43      79.63       4.98       0.00     0.00  380324.86     462.26  611170.01
00:12:05.841  
 Job: Malloc2p1 (Core Mask 0x2)
00:12:05.841  	 Verification LBA range: start 0x20 length 0x20
00:12:05.841  	 Malloc2p1           :       5.44      73.52       4.60       0.00     0.00  413089.06     460.31  647121.19
00:12:05.841  
 Job: Malloc2p2 (Core Mask 0x1)
00:12:05.841  	 Verification LBA range: start 0x0 length 0x20
00:12:05.841  	 Malloc2p2           :       5.43      79.62       4.98       0.00     0.00  379055.69     442.76  599186.29
00:12:05.841  
 Job: Malloc2p2 (Core Mask 0x2)
00:12:05.841  	 Verification LBA range: start 0x20 length 0x20
00:12:05.841  	 Malloc2p2           :       5.44      73.52       4.59       0.00     0.00  411753.58     481.77  639132.04
00:12:05.841  
 Job: Malloc2p3 (Core Mask 0x1)
00:12:05.841  	 Verification LBA range: start 0x0 length 0x20
00:12:05.841  	 Malloc2p3           :       5.43      79.61       4.98       0.00     0.00  377807.34     462.26  591197.14
00:12:05.841  
 Job: Malloc2p3 (Core Mask 0x2)
00:12:05.841  	 Verification LBA range: start 0x20 length 0x20
00:12:05.841  	 Malloc2p3           :       5.44      73.51       4.59       0.00     0.00  410415.31     464.21  631142.89
00:12:05.841  
 Job: Malloc2p4 (Core Mask 0x1)
00:12:05.841  	 Verification LBA range: start 0x0 length 0x20
00:12:05.841  	 Malloc2p4           :       5.48      81.81       5.11       0.00     0.00  367290.98     466.16  579213.41
00:12:05.841  
 Job: Malloc2p4 (Core Mask 0x2)
00:12:05.841  	 Verification LBA range: start 0x20 length 0x20
00:12:05.841  	 Malloc2p4           :       5.44      73.50       4.59       0.00     0.00  409074.38     456.41  619159.16
00:12:05.841  
 Job: Malloc2p5 (Core Mask 0x1)
00:12:05.841  	 Verification LBA range: start 0x0 length 0x20
00:12:05.841  	 Malloc2p5           :       5.48      81.80       5.11       0.00     0.00  366107.94     454.46  571224.26
00:12:05.841  
 Job: Malloc2p5 (Core Mask 0x2)
00:12:05.841  	 Verification LBA range: start 0x20 length 0x20
00:12:05.841  	 Malloc2p5           :       5.44      73.49       4.59       0.00     0.00  407609.71     454.46  607175.44
00:12:05.841  
 Job: Malloc2p6 (Core Mask 0x1)
00:12:05.841  	 Verification LBA range: start 0x0 length 0x20
00:12:05.841  	 Malloc2p6           :       5.48      81.80       5.11       0.00     0.00  364866.02     468.11  559240.53
00:12:05.841  
 Job: Malloc2p6 (Core Mask 0x2)
00:12:05.841  	 Verification LBA range: start 0x20 length 0x20
00:12:05.841  	 Malloc2p6           :       5.44      73.49       4.59       0.00     0.00  406323.15     448.61  599186.29
00:12:05.841  
 Job: Malloc2p7 (Core Mask 0x1)
00:12:05.841  	 Verification LBA range: start 0x0 length 0x20
00:12:05.841  	 Malloc2p7           :       5.48      81.79       5.11       0.00     0.00  363688.44     450.56  551251.38
00:12:05.841  
 Job: Malloc2p7 (Core Mask 0x2)
00:12:05.841  	 Verification LBA range: start 0x20 length 0x20
00:12:05.841  	 Malloc2p7           :       5.44      73.48       4.59       0.00     0.00  404969.47     450.56  591197.14
00:12:05.841  
 Job: TestPT (Core Mask 0x1)
00:12:05.841  	 Verification LBA range: start 0x0 length 0x100
00:12:05.841  	 TestPT              :       5.72      92.63       5.79       0.00     0.00 1253021.94   33704.23 1509949.44
00:12:05.841  
 Job: TestPT (Core Mask 0x2)
00:12:05.841  	 Verification LBA range: start 0x100 length 0x100
00:12:05.841  	 TestPT              :       5.67      87.43       5.46       0.00     0.00 1329744.82   27088.21 1677721.60
00:12:05.841  
 Job: raid0 (Core Mask 0x1)
00:12:05.841  	 Verification LBA range: start 0x0 length 0x200
00:12:05.841  	 raid0               :       5.62     106.89       6.68       0.00     0.00 1085023.64     819.20 1597830.10
00:12:05.841  
 Job: raid0 (Core Mask 0x2)
00:12:05.841  	 Verification LBA range: start 0x200 length 0x200
00:12:05.841  	 raid0               :       5.55      96.63       6.04       0.00     0.00 1202367.85     795.79 1765602.26
00:12:05.841  
 Job: concat0 (Core Mask 0x1)
00:12:05.841  	 Verification LBA range: start 0x0 length 0x200
00:12:05.841  	 concat0             :       5.68     112.71       7.04       0.00     0.00 1012573.80     780.19 1533916.89
00:12:05.841  
 Job: concat0 (Core Mask 0x2)
00:12:05.841  	 Verification LBA range: start 0x200 length 0x200
00:12:05.841  	 concat0             :       5.67      98.69       6.17       0.00     0.00 1156354.14     815.30 1701689.05
00:12:05.841  
 Job: raid1 (Core Mask 0x1)
00:12:05.841  	 Verification LBA range: start 0x0 length 0x100
00:12:05.841  	 raid1               :       5.70     137.58       8.60       0.00     0.00  827411.78    1006.45 1477992.84
00:12:05.841  
 Job: raid1 (Core Mask 0x2)
00:12:05.841  	 Verification LBA range: start 0x100 length 0x100
00:12:05.841  	 raid1               :       5.68     115.57       7.22       0.00     0.00  982657.96    1022.05 1637775.85
00:12:05.841  
 Job: AIO0 (Core Mask 0x1)
00:12:05.841  	 Verification LBA range: start 0x0 length 0x4e
00:12:05.841  	 AIO0                :       5.70     114.19       7.14       0.00     0.00  601517.37     643.66  798915.05
00:12:05.841  
 Job: AIO0 (Core Mask 0x2)
00:12:05.841  	 Verification LBA range: start 0x4e length 0x4e
00:12:05.841  	 AIO0                :       5.70     128.15       8.01       0.00     0.00  535053.39      85.33  874811.98
00:12:05.841  
 ===================================================================================================================
00:12:05.841  
 Total                       :               4025.74     251.61       0.00     0.00  574872.15      85.33 1973320.17
00:12:05.841  [2023-11-28 08:53:01.479325] thread.c:1347:spdk_thread_send_msg: *ERROR*: Thread TestPT is marked as exited.
00:12:05.841  bdevperf: thread.c:2643: spdk_for_each_channel_continue: Assertion `rc == 0' failed.
00:12:13.962  /home/vagrant/spdk_repo/spdk/test/common/autotest_common.sh: line 1062: 167724 Aborted                 (core dumped) "$@"
00:12:13.962    08:53:09	-- common/autotest_common.sh@1084 -- # trap - ERR
00:12:13.962    08:53:09	-- common/autotest_common.sh@1084 -- # print_backtrace
00:12:13.962    08:53:09	-- common/autotest_common.sh@1112 -- # [[ ehxBET =~ e ]]
00:12:13.962    08:53:09	-- common/autotest_common.sh@1114 -- # args=("${BASH_ARGV[@]}")
00:12:13.962    08:53:09	-- common/autotest_common.sh@1114 -- # local args
00:12:13.962    08:53:09	-- common/autotest_common.sh@1116 -- # xtrace_disable
00:12:13.962    08:53:09	-- common/autotest_common.sh@10 -- # set +x
00:12:13.962  ========== Backtrace start: ==========
00:12:13.962  
00:12:13.962  in /home/vagrant/spdk_repo/spdk/test/common/autotest_common.sh:1084 -> run_test(["bdev_verify_big_io"],["/home/vagrant/spdk_repo/spdk/build/examples/bdevperf"],["--json"],["/home/vagrant/spdk_repo/spdk/test/bdev/bdev.json"],["-q"],["128"],["-o"],["65536"],["-w"],["verify"],["-t"],["5"],["-C"],["-m"],["0x3"],[""])
00:12:13.962       ...
00:12:13.962     1079		timing_enter $test_name
00:12:13.962     1080		echo "************************************"
00:12:13.962     1081		echo "START TEST $test_name"
00:12:13.962     1082		echo "************************************"
00:12:13.962     1083		xtrace_restore
00:12:13.962     1084		time "$@"
00:12:13.962     1085		xtrace_disable
00:12:13.962     1086		echo "************************************"
00:12:13.962     1087		echo "END TEST $test_name"
00:12:13.962     1088		echo "************************************"
00:12:13.962     1089		timing_exit $test_name
00:12:13.962       ...
00:12:13.962  in /home/vagrant/spdk_repo/spdk/test/bdev/blockdev.sh:776 -> main([])
00:12:13.962       ...
00:12:13.962     771 	fi
00:12:13.962     772 	
00:12:13.962     773 	trap "cleanup" SIGINT SIGTERM EXIT
00:12:13.962     774 	
00:12:13.962     775 	run_test "bdev_verify" $rootdir/build/examples/bdevperf --json "$conf_file" -q 128 -o 4096 -w verify -t 5 -C -m 0x3 "$env_ctx"
00:12:13.962  => 776 	run_test "bdev_verify_big_io" $rootdir/build/examples/bdevperf --json "$conf_file" -q 128 -o 65536 -w verify -t 5 -C -m 0x3 "$env_ctx"
00:12:13.962     777 	run_test "bdev_write_zeroes" $rootdir/build/examples/bdevperf --json "$conf_file" -q 128 -o 4096 -w write_zeroes -t 1 "$env_ctx"
00:12:13.962     778 	
00:12:13.962     779 	# test json config not enclosed with {}
00:12:13.962     780 	run_test "bdev_json_nonenclosed" $rootdir/build/examples/bdevperf --json "$nonenclosed_conf_file" -q 128 -o 4096 -w write_zeroes -t 1 "$env_ctx" || true
00:12:13.962     781 	
00:12:13.962       ...
00:12:13.962  
00:12:13.962  ========== Backtrace end ==========
```

bt from the `bdevperf`'s core:

```
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib64/libthread_db.so.1".
Core was generated by `/home/vagrant/spdk_repo/spdk/build/examples/bdevperf --json /home/vagrant/spdk_'.
Program terminated with signal SIGABRT, Aborted.
#0  0x00007fba68d8baff in raise () from /usr/lib64/libc.so.6
[Current thread is 1 (Thread 0x7fba6cade440 (LWP 167724))]

Thread 3 (Thread 0x7fba628ff700 (LWP 167725)):
#0  0x00007fba68e6d027 in epoll_wait () from /usr/lib64/libc.so.6
No symbol table info available.
#1  0x000000000092d523 in eal_intr_handle_interrupts (pfd=6, totalfds=1) at ../lib/eal/linux/eal_interrupts.c:1077
        events = {{events = 1653592400, data = {ptr = 0x6ba0e21a00007fba, fd = 32698, u32 = 32698, u64 = 7755447159629053882}}}
        nfds = 0
#2  0x000000000092d9f6 in eal_intr_thread_main (arg=0x0) at ../lib/eal/linux/eal_interrupts.c:1163
        pipe_event = {events = 3, data = {ptr = 0x4, fd = 4, u32 = 4, u64 = 4}}
        src = 0x0
        numfds = 1
        pfd = 6
        __func__ = "eal_intr_thread_main"
#3  0x00000000008dfff3 in ctrl_thread_start (arg=0x60300000cca0) at ../lib/eal/common/eal_common_thread.c:285
        params = 0x60300000cca0
        start_arg = 0x0
        start_routine = 0x92d5e8 <eal_intr_thread_main>
#4  0x00007fba6910b1cf in start_thread () from /usr/lib64/libpthread.so.0
No symbol table info available.
#5  0x00007fba68d76e73 in clone () from /usr/lib64/libc.so.6
No symbol table info available.

Thread 2 (Thread 0x7fba61ffb700 (LWP 167726)):
#0  0x00007fba68e0d431 in __memset_avx2_erms () from /usr/lib64/libc.so.6
No symbol table info available.
#1  0x00007fba6b9ada5f in __interceptor_memset.part.40 () from /usr/lib64/libasan.so.5
No symbol table info available.
#2  0x00000000008e6b75 in malloc_elem_free (elem=0x200038000000) at ../lib/eal/common/malloc_elem.c:598
        ptr = 0x20003804f600
        data_len = 1048960
#3  0x00000000008ea5e9 in malloc_heap_free (elem=0x20003804f580) at ../lib/eal/common/malloc_heap.c:895
        heap = 0x16b0940 <early_mem_config+19136>
        start = 0x200034800000
        aligned_start = 0x200034800000
        end = 0x41b58ab3
        aligned_end = 0xf5a700
        len = 2097152
        aligned_len = 2097152
        page_sz = 2097152
        msl = 0x16abf08 <early_mem_config+136>
        i = 1
        n_segs = 1360
        before_space = 0
        after_space = 0
        ret = 0
        unmapped = false
        internal_conf = 0x16b31c0 <internal_config>
        asan_ptr = 0x20003804f600
        asan_data_len = 1048960
#4  0x00000000008ed909 in mem_free (addr=0x20003804f600, trace_ena=true) at ../lib/eal/common/rte_malloc.c:37
No locals.
#5  0x00000000008ed9a3 in rte_free (addr=0x20003804f600) at ../lib/eal/common/rte_malloc.c:44
No locals.
#6  0x00000000008ca3ad in rte_memzone_free (mz=0x200000000a68) at ../lib/eal/common/eal_common_memzone.c:336
        name = "RG_ring_28_167724", '\000' <repeats 14 times>
        mcfg = 0x16abe80 <early_mem_config>
        arr = 0x16abea8 <early_mem_config+40>
        found_mz = 0x200000000a68
        ret = 0
        addr = 0x20003804f600
        idx = 37
#7  0x0000000000b2e878 in rte_ring_free (r=0x20003804f600) at ../lib/ring/rte_ring.c:354
        ring_list = 0x16b0488 <early_mem_config+17928>
        te = 0x200000200f40
#8  0x00000000006efc88 in spdk_ring_free (ring=0x20003804f600) at env.c:396
No locals.
#9  0x00000000007eeca2 in _free_thread (thread=0x618000007480) at thread.c:415
        ch = 0x0
        msg = 0x0
        poller = 0x0
        ptmp = 0x73c153 <_reactor_remove_lw_thread+594>
        __func__ = "_free_thread"
        __PRETTY_FUNCTION__ = "_free_thread"
#10 0x00000000007f082a in spdk_thread_destroy (thread=0x618000007480) at thread.c:737
        __PRETTY_FUNCTION__ = "spdk_thread_destroy"
        __func__ = "spdk_thread_destroy"
#11 0x000000000073c2c3 in reactor_post_process_lw_thread (reactor=0x61b000006540, lw_thread=0x6180000077c0) at reactor.c:870
        thread = 0x618000007480
#12 0x000000000073c720 in _reactor_run (reactor=0x61b000006540) at reactor.c:924
        thread = 0x618000007480
        lw_thread = 0x6180000077c0
        tmp = 0x0
        now = 1252398581022
        rc = 1
#13 0x000000000073ca33 in reactor_run (arg=0x61b000006540) at reactor.c:952
        reactor = 0x61b000006540
        thread = 0x7fba61ff8de0
        lw_thread = 0x7fba61ff8de0
        tmp = 0x7fba61ff92c0
        thread_name = "reactor_1\000\371\000\000\000\000\000$g\221", '\000' <repeats 12 times>
        last_sched = 0
        __func__ = "reactor_run"
#14 0x00000000008df83b in eal_thread_loop (arg=0x1) at ../lib/eal/common/eal_common_thread.c:212
        f = 0x73c7af <reactor_run>
        fct_arg = 0x61b000006540
        lcore_id = 1
        cpuset = "1\000\000\000\000\000\000\000\b\000\000\000\000\000\000\000\000\020\000\000\000\000\000\000\000\300\377a\272\177\000\000\000\000\200\000\000\000\000\000\000\000\001\000\060`\000\000 ", '\000' <repeats 41 times>, "\216\307euI\t", '\000' <repeats 16 times>, " \300\001c\272\177\000\000\230\221\377a\272\177\000\000\240\221\377a\272\177\000\000@\300\001c\272\177\000\000\000\360\017\000\000\000\000\000\000\200/\314\367\017\000\000\062@\000\000\000\000\000\000\264\001\241k\272\177\000\000\000\300\177a\272\177\000\000\000\000\216\307euI\t\000\360\377\377\377\377\377\377"...
        ret = 0
#15 0x000000000091c005 in eal_worker_thread_loop (arg=0x1) at ../lib/eal/linux/eal.c:916
No locals.
#16 0x00007fba6910b1cf in start_thread () from /usr/lib64/libpthread.so.0
No symbol table info available.
#17 0x00007fba68d76e73 in clone () from /usr/lib64/libc.so.6
No symbol table info available.

Thread 1 (Thread 0x7fba6cade440 (LWP 167724)):
#0  0x00007fba68d8baff in raise () from /usr/lib64/libc.so.6
No symbol table info available.
#1  0x00007fba68d5eea5 in abort () from /usr/lib64/libc.so.6
No symbol table info available.
#2  0x00007fba68d5ed79 in __assert_fail_base.cold.0 () from /usr/lib64/libc.so.6
No symbol table info available.
#3  0x00007fba68d84456 in __assert_fail () from /usr/lib64/libc.so.6
No symbol table info available.
#4  0x00000000007fb8d3 in spdk_for_each_channel_continue (i=0x60700002a100, status=0) at thread.c:2643
        thread = 0x0
        ch = 0x0
        dev = 0x613000007580
        rc = -5
        __PRETTY_FUNCTION__ = "spdk_for_each_channel_continue"
#5  0x0000000000793a38 in spdk_bdev_for_each_channel_continue (iter=0x603000020fb0, status=0) at bdev.c:10023
No locals.
#6  0x000000000076c258 in bdev_channel_submit_qos_io (i=0x603000020fb0, bdev=0x61600000b480, io_ch=0x61200003ebc0, ctx=0x611000017380) at bdev.c:3726
        bdev_ch = 0x61200003ec20
        status = 0
#7  0x0000000000793b72 in bdev_each_channel_msg (i=0x60700002a100) at bdev.c:10042
        iter = 0x603000020fb0
        bdev = 0x61600000b480
        ch = 0x61200003ebc0
#8  0x00000000007fad40 in _call_channel (ctx=0x60700002a100) at thread.c:2529
        i = 0x60700002a100
        ch = 0x61200003ebc0
#9  0x00000000007f0ecd in msg_queue_run_batch (thread=0x618000007080, max_msgs=8) at thread.c:841
        msg = 0x200004490a00
        count = 8
        i = 0
        messages = {0x200004490a00, 0x200004490940, 0x200004490880, 0x2000044907c0, 0x200004490700, 0x200004490640, 0x200004490580, 0x2000044a7440}
        notify = 1
        rc = 32766
        __func__ = "msg_queue_run_batch"
        __PRETTY_FUNCTION__ = "msg_queue_run_batch"
#10 0x00000000007f29ea in thread_poll (thread=0x618000007080, max_msgs=0, now=1252398580308) at thread.c:1063
        msg_count = 0
        poller = 0x12398c5b700
        tmp = 0x618000001080
        critical_msg = 0x0
        rc = 0
#11 0x00000000007f3281 in spdk_thread_poll (thread=0x618000007080, max_msgs=0, now=1252398580308) at thread.c:1156
        orig_thread = 0x0
        rc = 0
#12 0x000000000073c596 in _reactor_run (reactor=0x61b000006280) at reactor.c:914
        thread = 0x618000007080
        lw_thread = 0x6180000073c0
        tmp = 0x0
        now = 1252398580308
        rc = 0
#13 0x000000000073ca33 in reactor_run (arg=0x61b000006280) at reactor.c:952
        reactor = 0x61b000006280
        thread = 0x7ffe6ba2f3e0
        lw_thread = 0x20000402e480
        tmp = 0x7ffe6ba2f380
        thread_name = "reactor_0\000\213\000\000\000\000\000g\332r\000\000\000\000\000\200\020\000\000\177\000\000"
        last_sched = 0
        __func__ = "reactor_run"
#14 0x000000000073d3f5 in spdk_reactors_start () at reactor.c:1068
        reactor = 0x61b000006280
        i = 4294967295
        current_core = 0
        rc = 0
        __func__ = "spdk_reactors_start"
        __PRETTY_FUNCTION__ = "spdk_reactors_start"
#15 0x0000000000732db1 in spdk_app_start (opts_user=0x7ffe6ba2f810, start_fn=0x41d1bc <bdevperf_run>, arg1=0x0) at app.c:835
        rc = 0
        tty = 0x0
        tmp_cpumask = {str = '\000' <repeats 256 times>, cpus = "\001", '\000' <repeats 126 times>}
        g_env_was_setup = false
        opts_local = {name = 0xe33860 "bdevperf", json_config_file = 0x7ffe6ba30df9 "/home/vagrant/spdk_repo/spdk/test/bdev/bdev.json", json_config_ignore_errors = false, reserved17 = "\000\000\000\000\000\000", rpc_addr = 0x0, reactor_mask = 0x7ffe6ba30e4f "0x3", tpoint_group_mask = 0x0, shm_id = -1, reserved52 = "\000\000\000", shutdown_cb = 0x41d505 <spdk_bdevperf_shutdown_cb>, enable_coredump = true, reserved65 = "\000\000", mem_channel = -1, main_core = -1, mem_size = -1, no_pci = false, hugepage_single_segments = false, unlink_hugepage = false, reserved83 = "\000\000\000\000", hugedir = 0x0, print_level = SPDK_LOG_INFO, reserved100 = "\000\000\000", num_pci_addr = 0, pci_blocked = 0x0, pci_allowed = 0x0, iova_mode = 0x0, delay_subsystem_init = false, reserved137 = "\000\000\000\000\000\000", num_entries = 32768, env_context = 0x0, log = 0x0, base_virtaddr = 35184372088832, opts_size = 236, disable_signal_handlers = false, interrupt_mode = false, reserved186 = "\000\000\000\000\000", msg_mempool_size = 262143, rpc_allowlist = 0x0, vf_token = 0x0, lcore_map = 0x0, rpc_log_level = SPDK_LOG_DISABLED, rpc_log_file = 0x0}
        opts = 0x7ffe6ba2f4b0
        i = 128
        __func__ = "spdk_app_start"
#16 0x000000000041eb25 in main (argc=15, argv=0x7ffe6ba2fa28) at bdevperf.c:2712
        opts = {name = 0xe33860 "bdevperf", json_config_file = 0x7ffe6ba30df9 "/home/vagrant/spdk_repo/spdk/test/bdev/bdev.json", json_config_ignore_errors = false, reserved17 = "\000\000\000\000\000\000", rpc_addr = 0x0, reactor_mask = 0x7ffe6ba30e4f "0x3", tpoint_group_mask = 0x0, shm_id = -1, reserved52 = "\000\000\000", shutdown_cb = 0x41d505 <spdk_bdevperf_shutdown_cb>, enable_coredump = true, reserved65 = "\000\000", mem_channel = -1, main_core = -1, mem_size = -1, no_pci = false, hugepage_single_segments = false, unlink_hugepage = false, reserved83 = "\000\000\000\000", hugedir = 0x0, print_level = SPDK_LOG_INFO, reserved100 = "\000\000\000", num_pci_addr = 0, pci_blocked = 0x0, pci_allowed = 0x0, iova_mode = 0x0, delay_subsystem_init = false, reserved137 = "\000\000\000\000\000\000", num_entries = 32768, env_context = 0x0, log = 0x0, base_virtaddr = 35184372088832, opts_size = 236, disable_signal_handlers = false, interrupt_mode = false, reserved186 = "\000\000\000\000\000", msg_mempool_size = 0, rpc_allowlist = 0x0, vf_token = 0x0, lcore_map = 0x0, rpc_log_level = SPDK_LOG_DISABLED, rpc_log_file = 0x0}
        rc = 1
```

**Comment by tomzawadzki** [comment]
This seems like a result of following patch https://review.spdk.io/gerrit/c/spdk/spdk/+/18914.
Summing up the above, the `bdev_channel_poll_qos` poller now calls `spdk_bdev_for_each_channel()` to submit the queued I/O.
In bdevperf once the job ends, all threads are exited. This should complete only once there is no longer anything going on - see `thread_exit()` conditions. Yet per this report it seems that once all io_channels were iterated over, the original thread was already exited:
https://github.com/spdk/spdk/blob/fde1276b1e77c46e25749ce81246764bbf6ced39/lib/thread/thread.c#L2642-L2643

Other than reverting the patch, maybe preventing the thread exit (or poller unregister) before the `spdk_for_each_channel()` or `spdk_bdev_for_each_channel()` finishes would be in order.

Let's discuss on bug scrub.

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @kgodzwox. log: https://ci.spdk.io/public_build/autotest-per-patch_112937.html

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @kgodzwox. log: https://ci.spdk.io/public_build/autotest-per-patch_112947.html

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @kgodzwox. log: https://ci.spdk.io/public_build/autotest-per-patch_112943.html

**Comment by shuhei-matsumoto** [comment]
Hi, I will push patches for review today.

**Comment by shuhei-matsumoto** [comment]
I hit this issue very often. I wrote a patch for review.
https://review.spdk.io/gerrit/c/spdk/spdk/+/20788
Any feedback is very appreciated.

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @KKaras169169. log: https://ci.spdk.io/public_build/autotest-per-patch_112976.html

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @KKaras169169. log: https://ci.spdk.io/public_build/autotest-per-patch_112986.html

**Comment by spdkci** [comment]
Another instance of this failure. Reported by @KKaras169169. log: https://ci.spdk.io/public_build/autotest-per-patch_112972.html
